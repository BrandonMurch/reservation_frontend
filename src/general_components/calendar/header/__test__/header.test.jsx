// Dependencies
import React from 'react';
import { unmountComponentAtNode } from 'react-dom';
import { render, fireEvent } from '@testing-library/react';
import { create } from 'react-test-renderer';

// Components
import Header from '../index';

describe('<Header />', () => {
  let component;
  let container = null;
  const props = {
    date: 'August 2020',
    prev: jest.fn(),
    next: jest.fn(),
    isThisToday: false,
    goToToday: jest.fn(),

  };

  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
    component = render(<Header {...props} />);
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it('should match snapshot', () => {
    const tree = create(<Header {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should display current month', () => {
    const month = component.getByText('August 2020');
    expect(month).toBeInTheDocument();
  });

  it('call functions with buttons', () => {
    let button = component.getByRole('button', { name: 'Today' });
    fireEvent.click(button);
    expect(props.goToToday).toHaveBeenCalled();
    button = component.getByRole('button', { name: '>' });
    fireEvent.click(button);
    expect(props.next).toHaveBeenCalled();
    button = component.getByRole('button', { name: '<' });
    fireEvent.click(button);
    expect(props.prev).toHaveBeenCalled();
  });
});
