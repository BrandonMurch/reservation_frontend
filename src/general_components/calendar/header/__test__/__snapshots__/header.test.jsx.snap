// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshot 1`] = `
<div
  className="container"
>
  <div
    className="monthTextContainer"
  >
    <button
      aria-label="date-picker"
      className="monthText"
      onClick={[Function]}
      type="button"
    >
      August 2020
    </button>
  </div>
  <div
    className="buttonContainer"
  >
    <button
      className="button"
      onClick={[MockFunction]}
      type="button"
    >
      Today
    </button>
    <button
      aria-label="previous-date"
      className="prevButton"
      onClick={[MockFunction]}
      type="button"
    >
      ➤
    </button>
    <button
      aria-label="next-date"
      className="button"
      onClick={[MockFunction]}
      type="button"
    >
      ➤
    </button>
  </div>
</div>
`;
